arangod/Replication2/ReplicatedLog/Components/IStorageManager.h

```diff
@@ -57,7 +57,7 @@ struct IStorageTransaction {
       -> futures::Future<Result> = 0;
   virtual auto removeBack(LogIndex start) noexcept
       -> futures::Future<Result> = 0;
-  virtual auto appendEntries(InMemoryLog) noexcept
+  virtual auto appendEntries(InMemoryLog, bool waitForSync) noexcept
```

Comment:
I'm totally fine with having the `bool` parameter in here. But arangodb code tends to have 4 or 5 of these parameters on functions to make them _just behave a little different_.

This is why I added the `WriteOptions` to the `IStorageEngineMethods` interface. This allows to write code that looks similar to python code with key word arguments. In particular you don't have to specify the first 4 if you want to change the default value of the fifth one.

Reply:
I don't have a strong preference on this, but I'm happy to make it look more similar to Python :)